Extensions/Video/JsExtension.js

```diff
@@ -0,0 +1,775 @@
+   */
+  runExtensionSanityTests: function(gd, extension) {
+    return [];
+  },
+  /**
+   * Register editors for objects.
+   *
+   * ℹ️ Run `node import-GDJS-Runtime.js` (in newIDE/app/scripts) if you make any change.
+   */
+
+  registerEditorConfigurations: function(objectsEditorService) {
+    objectsEditorService.registerEditorConfiguration(
+      "Video::VideoObject",
+      objectsEditorService.getDefaultObjectJsImplementationPropertiesEditor()
+    );
+  },
+  /**
+   * Register renderers for instance of objects on the scene editor.
+   *
+   * ℹ️ Run `node import-GDJS-Runtime.js` (in newIDE/app/scripts) if you make any change.
+   */
+  registerInstanceRenderers: function(objectsRenderingService) {
+    const RenderedInstance = objectsRenderingService.RenderedInstance;
+    const PIXI = objectsRenderingService.PIXI;
+
+    /**
+     * Renderer for instances of VideoObject inside the IDE.
+     *
+     * @extends RenderedInstance
+     * @class RenderedVideoObjectInstance
+     * @constructor
+     */
+    function RenderedVideoObjectInstance(
+      project,
+      layout,
+      instance,
+      associatedObject,
+      pixiContainer,
+      pixiResourcesLoader
+    ) {
+      RenderedInstance.call(
+        this,
+        project,
+        layout,
+        instance,
+        associatedObject,
+        pixiContainer,
+        pixiResourcesLoader
+      );
+
+      //TODO Need ressource manager and can select only video files
+      //Codec can be played : https://www.w3schools.com/tags/av_met_canplaytype.asp
+
+      var textureVideo = new PIXI.Texture.fromVideo(
+        "C:/Users/RTX-Bouh/Desktop/test_video_GD.mp4"
+      );
+
+      //FIXME This autoplay don't work
+      textureVideo.baseTexture.source.autoplay = false;
+
+      console.log(textureVideo);
+
+      //Setup the PIXI object:
+      this._pixiObject = new PIXI.Sprite(textureVideo);
+      this._pixiObject.anchor.x = 0.5;
+      this._pixiObject.anchor.y = 0.5;
+      this._pixiContainer.addChild(this._pixiObject);
+      this.update();
+    }
+    RenderedVideoObjectInstance.prototype = Object.create(
+      RenderedInstance.prototype
+    );
+
+    /**
+     * Return the path to the thumbnail of the specified object.
+     */
+    RenderedVideoObjectInstance.getThumbnail = function(
+      project,
+      resourcesLoader,
+      object
+    ) {
+      return "JsPlatform/Extensions/videoicon24.png";
+    };
+
+    /**
+     * This is called to update the PIXI object on the scene editor
+     */
+    RenderedVideoObjectInstance.prototype.update = function() {
+      //FIXME need help here i can't catch err_file_not_found like this :
+      video_is_ok = true;
+      this._pixiObject._texture.baseTexture.error = function() {
+        console.log("Starting to load video");
+        console.log("Force pause");
+        //this._pixiObject._texture.baseTexture.source.pause();
+        video_is_ok = false;
+      };
+
+      if (video_is_ok == false) {
```

Comment:
This is not working because video_is_ok is always trye. `onError` will be called later, possibly milliseconds or even seconds later. So `video_is_ok` will ALWAYS be true here.

Check articles on Google about "Javascript asynchronous callbacks". This one looks good: https://blog.codeanalogies.com/2016/04/11/javascript-callbacks-explained-using-minions/ :)

Reply:
So the content of this `if` should be placed INSIDE `onError` function callback.